articlequality/feature_lists/ptwiki.py

```diff
@@ -0,0 +1,169 @@
+    "|".join(MAIN_TEMPLATES), name="ptwiki.main_article_templates")
+CITE_TEMPLATES = [
+    r"Cite",
+    r"Citar",
+    r"Citar web",
+    r"Citar livro",
+    r"Harvard[_ ]citation[_ ]no[_ ]brackets", r"harvnb",
+    r"Harvard citation", r"harv",
+    r"Harvard citation text", r"harvtxt",
+    r"Harvcoltxt",
+    r"Harvcol",
+    r"Harvcolnb",
+    r"Harvard citations", r"harvs",
+    r"Harvp"
+]
+cite_templates = wikitext.revision.template_names_matching(
+    "|".join(CITE_TEMPLATES), name="ptwiki.revision.cite_templates")
+shortened_footnote_templates = wikitext.revision.template_names_matching(
+    r"sfn", name="ptwiki.revision.shortened_footnote_templates")
+all_ref_tags = shortened_footnote_templates + wikitext.revision.ref_tags
+all_cite_templates = cite_templates + shortened_footnote_templates
+proportion_of_templated_references = \
+    all_cite_templates / max(all_ref_tags, 1)
+non_templated_references = max(all_ref_tags - all_cite_templates, 0)
+non_cite_templates = sub(
+    wikitext.revision.templates, all_cite_templates,
+    name="ptwiki.revision.non_cite_templates"
+)
+
+# Links
+category_links = wikitext.revision.wikilink_titles_matching(
+    r"(Category|Categoria)\:", name="ptwiki.revision.category_links")
+
+image_links = wikitext.revision.wikilink_titles_matching(
+    r"(File|Ficheiro|Arquivo|Image|Imagem)\:",
+    name="ptwiki.revision.image_links")
+
+image_templates = wikitext.revision.template_names_matching(
+    r"(Scalable[ _]image|Panorama|Imagem[ _]vertical)|Panorama|Panorama 2",
+    name='ptwiki.revision.image_template')
+
+
+def get_images(strs):
+    """
+    Parses a list of strings, expected to be tags or templates
+    to get matching image patterns and returns a count of the matches.
+    """
+    matches = re.findall(
+        r"image[1-9]{1,2}|(File|Ficheiro|" +
+        r"Image|Imagem|Archivo):|photo[1-9][a-z]",
+        "".join(strs), re.I)
+    return len(matches)
+
+
+image_templates_str = wikitext.revision.datasources.templates_str_matching(
+    r"{{(imagem multipla|galeria|multiple image|image array|gallery)",
+    name='ptwiki.revision.image_templates_str')
+
+images_in_templates = Feature("ptwiki.revision.images_in_templates",
+                              get_images,
+                              depends_on=[image_templates_str],
+                              returns=int)
+
+image_tags_str = wikitext.revision.datasources.tags_str_matching(
+    r"<(gallery|imagemap)",
+    name='ptwiki.revision.image_tags_str')
+
+images_in_tags = Feature("ptwiki.revision.images_in_tags",
+                         get_images,
+                         depends_on=[image_tags_str],
+                         returns=int)
+
+
+def get_infobox_images(strs):
+    matches = re.findall(
+        r"\.(jpg|jpeg|png|gif|svg|tiff|pdf|ogg|djvu)", "".join(strs), re.I)
+    return len(matches)
+
+
+infobox_templates_str = wikitext.revision.datasources.templates_str_matching(
+    r"{{Infobox",
```

Comment:
Most pages on ptwiki use `{{Info` instead of `{{Infobox` (which the community deprecated)